File: Java/CommandRobotSkeleton/src/main/java/frc/robot/subsystems/TelescopicSubsystem.java
// Copyright (c) FIRST and other WPILib contributors.
// Open Source Software; you can modify and/or share it under the terms of
// the WPILib BSD license file in the root directory of this project.

package frc.robot.subsystems;

import edu.wpi.first.wpilibj2.command.SubsystemBase;

import com.revrobotics.spark.SparkMax;
import com.revrobotics.spark.SparkLowLevel.MotorType;

import frc.robot.RobotID;

public class TelescopicSubsystem extends SubsystemBase {
    /** Creates a new TelescopicSubsystem. */

    private SparkMax telescopicMotor;


    public TelescopicSubsystem() {
        telescopicMotor = new SparkMax(RobotID.TELESCOPIC_ID, MotorType.kBrushless);
    }


    public void extendTelescopic(double power) {
        telescopicMotor.set(power);
    }

    public void shortenTelescopic(double power) {
        telescopicMotor.set(-power);
    }

    @Override
    public void periodic() {
        // This method will be called once per scheduler run
    }
}

File: Cpp/CommandRobotSkeleton/src/main/cpp/subsystems/TelescopicSubsystem.cpp
#include "subsystems/TelescopicSubsystem.h"
#include "RobotID.h" // Constants.h yerine yeni dosyamızı ekledik

TelescopicSubsystem::TelescopicSubsystem()
    : m_telescopicMotor(RobotID::TELESCOPIC_ID, rev::spark::SparkLowLevel::MotorType::kBrushless) {}

void TelescopicSubsystem::ExtendTelescopic(double power) {
  m_telescopicMotor.Set(power);
}

void TelescopicSubsystem::ShortenTelescopic(double power) {
  m_telescopicMotor.Set(-power);
}

void TelescopicSubsystem::Periodic() {
  // Bu metod scheduler her çalıştığında bir kez çağrılır
}